libsql-server/src/connection/libsql.rs

```diff
@@ -76,60 +60,62 @@ where
     ) -> Result<Self> {
         let txn_timeout = config_store.get().txn_timeout.unwrap_or(TXN_TIMEOUT);
 
-        let mut this = Self {
-            db_path,
-            stats,
-            broadcaster,
-            config_store,
-            extensions,
-            max_response_size,
-            max_total_response_size,
-            auto_checkpoint,
-            current_frame_no_receiver,
-            _db: None,
-            wal_wrapper,
-            encryption_config,
-            block_writes,
-            resolve_attach_path,
-            connection_manager: ConnectionManager::new(txn_timeout),
-            make_wal_manager,
-        };
-
-        let db = this.try_create_db().await?;
-        this._db = Some(db);
-
-        Ok(this)
-    }
+        let (snd, recv) = async_channel::bounded(16);
+        tokio::spawn(async move {
+            let inner = MakeLibSqlConnInner {
+                db_path,
+                stats,
+                broadcaster,
+                config_store,
+                extensions,
+                max_response_size,
+                max_total_response_size,
+                auto_checkpoint,
+                current_frame_no_receiver,
+                wal_wrapper,
+                encryption_config,
+                block_writes,
+                resolve_attach_path,
+                connection_manager: ConnectionManager::new(txn_timeout),
+                make_wal_manager,
+            };
 
-    /// Tries to create a database, retrying if the database is busy.
-    async fn try_create_db(&self) -> Result<LibSqlConnection<W>> {
-        // try 100 times to acquire initial db connection.
-        let mut retries = 0;
-        loop {
-            match self.make_connection().await {
-                Ok(conn) => return Ok(conn),
-                Err(
-                    err @ Error::RusqliteError(rusqlite::Error::SqliteFailure(
-                        rusqlite::ffi::Error {
-                            code: ErrorCode::DatabaseBusy,
-                            ..
-                        },
-                        _,
-                    )),
-                ) => {
-                    if retries < 100 {
-                        tracing::warn!("Database file is busy, retrying...");
-                        retries += 1;
-                        tokio::time::sleep(Duration::from_millis(100)).await
-                    } else {
-                        Err(err)?;
-                    }
+            loop {
```

Comment:
Does sqld have some graceful shutdown logic? Should we stop this coroutine at some point and reclaim all resources?

Reply:
Yes, and this contouring will exit when we drop all the receivers, this will happen when the namespace is deallocated